### PR TITLE
Correctly handle hosts with port number

### LINKED ISSUE #2598: No root pages with port numbers
**Affected version(s)**

Contao 4.10.5 (not testet in earlier versions of 4.10. branch)

**Description**
It is not possible to open the website if you enter a port number:
Bildschirmfoto 2020-12-28 um 11 38 00
If you add an port number in url field it will be stripped. I'm using https (:8890) under a local environment.

If you enter `sub.test.local:8890` it appears:
Bildschirmfoto 2020-12-28 um 11 32 31

If you enter `sub.test.local` it appears:
Bildschirmfoto 2020-12-28 um 10 53 19

No error logs will produced
PHP 7.4.9

My mamp config:
Bildschirmfoto 2020-12-28 um 11 44 58

See discussion here:
[https://contao.slack.com/archives/CJT61P8R1/p1609147484351900](https://contao.slack.com/archives/CJT61P8R1/p1609147484351900)

Composer.json:
```
{
    "type": "project",
    "require": {
        "alnv/catalog-manager": "^1.31",
        "bugbuster/contao-statistic_group-bundle": "^1.0",
        "bugbuster/contao-visitors-bundle": "^1.6",
        "comolo/contao-supertheme": "^3.3",
        "contao/conflicts": "*@dev",
        "contao/manager-bundle": "^4.9",
        "fritzmg/contao-lazy-load-element": "^1.0",
        "fritzmg/contao-swiper": "^3.3",
        "heimrichhannot/contao-speed-bundle": "^1.9",
        "inspiredminds/contao-wowjs": "^1.0",
        "jrgregory/m17-sticky-backend-footer": "^2.0",
        "martin-kozianka/contao-timetags": "^1.0",
        "nerdlichter/contao-image-optimizer-bundle": "^1.0",
        "richardhj/contao-childlayouts": "^1.1",
        "terminal42/contao-inserttags": "^1.9",
        "terminal42/contao-node": "^1.3",
        "zmyslny/contao-wrapper_tags": "^2.1"
    },
    "extra": {
        "contao-component-dir": "assets"
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    }
}

```

### PR DESCRIPTION
**Pull request by leofeyer**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2598
| Docs PR or issue | -

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Environment.php [hunk 1]
```diff
@@ -467,7 +467,7 @@ protected static function base()
 	 */
 	protected static function host()
 	{
-		return static::get('httpHost');
+		return preg_replace('/:\d+$/', '', static::get('httpHost'));
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Environment.php:470]: This is a necessary BC break, otherwise there is no way to make an extension compatible with both Contao 4.9 (and below) and Contao 4.10 (and above). If someone explicitly needs the hostname with port number, they can use `Environment::get('httpHost')` or `{{env::http_host}}`, which also works in all Contao 4 versions. And since production URLs never have port numbers, only a few local website should be affected.

File: core-bundle/tests/Fixtures/Functional/Routing/localhost.yml [hunk 2]
```diff
@@ -3,11 +3,11 @@ tl_page:
...
       sorting: 128
       tstamp: 1539696285
-      title: Domain with port
-      alias: domain-with-port
+      title: Localhost
+      alias: localhost
       type: root
       language: en
-      dns: domain-with-port.local:8080
+      dns: 127.0.0.1
```
Comment (bytehead) [inline core-bundle/tests/Fixtures/Functional/Routing/localhost.yml:10]: Just curious - why did you change to an IP here?

Comment (leofeyer) [reply core-bundle/tests/Fixtures/Functional/Routing/localhost.yml:10]: Because that is the most common use case in combination with a port number. The corresponding functional test uses the URL `https://127.0.0.1:8080` as e.g. the Symfony local web server does.

**Review by bytehead — APPROVED** [review]
LGTM

**Review by aschempp — APPROVED** [review]
